src/lib/state-machine-api/state-machine.ts

```diff
@@ -0,0 +1,9 @@
+import { MachineState } from './machine-state';
+import { MachineEvent } from './machine-event';
+export interface StateMachine<TOutput> {
```

Comment:
its TOutput a generyc type? :thinking:

Reply:
Yes, its the inner state type of the machine, Mealy has it so that's why I added it